code/script.js

```diff
@@ -27,25 +39,196 @@ const showMessage = (message, sender) => {
+  console.log("select");
+  dropdown.id = 'food-dropdown';
+
+
+  dropdown.classList.add('dropdown-style');
+
+
+  foodOptions.forEach(option => {
+    console.log("option");
+    const optionElement = document.createElement('option');
+    optionElement.value = option;
+    optionElement.textContent = option;
+    dropdown.appendChild(optionElement);
+  });
+
+
+  dropdown.addEventListener('change', (event) => {
+    console.log("change");
+    const selectedfood = event.target.value;
+    showMessage(`Yummy! You've selected ${selectedfood}.`, 'bot');
+    showAgeOptions(selectedfood);
+  });
+
+  inputHandle.appendChild(dropdown);
+}
+
+
+function showAgeOptions(selectedSize) {
+  console.log(showAgeOptions);
+  showMessage(`Great ${userName}! What size do you like to order your food? Do you want a regular size or small size, may I ask if you are an adult or child?`, 'bot');
+  inputHandle.innerHTML = '';
+
+  const ageButtons = document.createElement('div');
+  console.log(ageButtons);
+  ageButtons.innerHTML = `
+  <button class="send-btn" id= "adult">Adult</button>
+  <button class="send-btn" id= "child">Child</button>
+  `;
+  inputHandle.appendChild(ageButtons);
+
+  const ageButtonsElements = ageButtons.querySelectorAll('.send-btn')
+  console.log(ageButtonsElements);
+  ageButtonsElements.forEach(button => {
+    button.addEventListener('click', (event) => {
+      const selectedSize = event.target.id;
+      showMessage(`You've selected the ${selectedSize} option!`, 'bot');
+      ageOptions(selectedSize);
+    });
+  });
+
+}
+
+function ageOptions(choice) {
+  console.log(ageOptions);
+  inputHandle.innerHTML = '';
+
+  if (choice === 'adult') {
+    showMessage(`The price for a adult size food is 15€.`, 'bot');
+  } else if (choice === 'child') {
+    showMessage(`The price for a small size food is 10€.`, 'bot');
+    orderConfirmation(choice);
+  };
+};
+
+// I have been struggling with this part a lot - I dont find the solution and the time is running out - sunday evening. 
+
+function orderConfirmation(foodChoice, selectedSize) {
+  showMessage(`This is your order, ${userName}: ${foodChoice} (${selectedSize} size).`, 'bot');
+
+  inputHandle.innerHTML = ` 
+    <button class="send-btn" id="confirm">Yes, I accept this order!</button>
+    <button class="send-btn" id="cancel">No, I want to cancel my order.</button>
+  `;
+
+  const confirmBtn = document.getElementById('confirm');
+  const cancelBtn = document.getElementById('cancel');
+
+  confirmBtn.addEventListener('click', () => {
+    showMessage(`Thanks for confirming your order! You ordered ${foodChoice} (${selectedSize} size). Your food will soon be at your place.`, 'bot');
+    inputHandle.innerHTML = '';
+  });
+
+  cancelBtn.addEventListener('click', () => {
+    showMessage(`Your order has been canceled. Welcome back! Feel free to order food next time you are hungry.`, 'bot');
+    inputHandle.innerHTML = '';
+  });
+}
```

Comment:
There is a bug here. It shows the selectedSize is not defined. The second parameter "selectedSize" doesn't pass any value. It is undefined

Reply:
Thank you Miko! 💫 I have now updated this repo.